ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.tsx

```diff
@@ -0,0 +1,201 @@
+import TypedMetacardDefs from '../tabs/metacard/metacardDefinitions'
+import Chip from '@material-ui/core/Chip'
+import Grid from '@material-ui/core/Grid'
+import Swath from '../swath/swath'
+import { Typography } from '@material-ui/core'
+const properties = require('../../js/properties.js')
+const metacardDefinitions = require('../singletons/metacard-definitions.js')
+import FilterInput from '../../react-component/filter/filter-input'
+export interface BasicFilterClass extends Omit<FilterClass, 'property'> {
+  property: string[]
+}
+
+type QueryTimeProps = {
+  value: undefined | BasicFilterClass
+  onChange: (e: any) => void
+}
+
+const getPossibleProperties = () => {
+  return metacardDefinitions.sortedMetacardTypes
+    .filter(
+      (definition: any) => !definition.hidden && definition.type === 'DATE'
+    )
+    .map((definition: any) => ({
+      label: definition.alias || definition.id,
+      value: definition.id,
+    })) as [
+    {
+      label: string
+      value: string
+    }
+  ]
+}
+
+const determinePropertiesToApplyTo = ({
+  value,
+}: {
+  value: BasicFilterClass
+}): Array<{ label: string; value: string }> => {
+  if (value.property) {
+    return value.property.map(property => {
+      return {
+        label: TypedMetacardDefs.getAlias({ attr: property }),
+        value: property,
+      }
+    })
+  } else {
+    return (properties.basicSearchTemporalSelectionDefault || []).map(
+      (property: string) => {
+        return {
+          label: TypedMetacardDefs.getAlias({ attr: property }),
+          value: property,
+        }
+      }
+    )
+  }
+}
+
+const QueryTime = ({ value, onChange }: QueryTimeProps) => {
+  React.useEffect(
+    () => {
+      if (value && value.property === undefined) {
+        onChange({
+          ...value,
+          property: determinePropertiesToApplyTo({ value }).map(
+            val => val.value
+          ),
+        })
+      }
+    },
+    [value]
+  )
+  if (value && value.property === undefined) {
+    return null // the use effect above should fire to take care of setting a default
+  }
+  return (
+    <>
+      <div>
+        <Typography className="pb-2">Time Range</Typography>
+
+        <TextField
+          fullWidth
+          variant="outlined"
+          size="small"
+          select
+          value={value ? value.type : 'any'}
+          onChange={e => {
+            if (e.target.value === 'any') {
+              onChange(undefined)
+            } else {
+              onChange({
+                ...value,
+                type: e.target.value,
+              })
+            }
+          }}
+        >
+          <MenuItem value="any">Any</MenuItem>
+          <MenuItem value="AFTER">After</MenuItem>
+          <MenuItem value="BEFORE">Before</MenuItem>
+          <MenuItem value="DURING">Between</MenuItem>
+          <MenuItem value="RELATIVE">Relative</MenuItem>
+        </TextField>
+      </div>
+      {value ? (
```

Comment:
Out of curiosity, is there a reason that you prefer a ternary to `{ value && (<Component />) }`?

Reply:
My personal preference, it weirds me out to use the ANDing logic that way.  Definitely more terse than the ternary, but to me a bit less readable since it feels like it's taking advantage of a side effect.